src/Nethermind/Nethermind.Evm/Metrics.cs

```diff
@@ -8,71 +8,87 @@ namespace Nethermind.Evm;
 
 public class Metrics
 {
-    [CounterMetric]
+    [GaugeMetric]
+    [Description("Number of EXTCODESIZE opcodes executed.")]
+    public static long ExtCodeSizeOpcode;
+
+    [GaugeMetric]
+    [Description("Number of EXTCODESIZE ISZERO optimizations.")]
+    public static long ExtCodeSizeOptimizedIsZero;
+
+    [GaugeMetric]
+    [Description("Number of EXTCODESIZE GT optimizations.")]
+    public static long ExtCodeSizeOptimizedGT;
+
+    [GaugeMetric]
+    [Description("Number of EXTCODESIZE EQ optimizations.")]
+    public static long ExtCodeSizeOptimizedEQ;
+
+    [GaugeMetric]
     [Description("Number of EVM exceptions thrown by contracts.")]
     public static long EvmExceptions { get; set; }
 
-    [CounterMetric]
+    [GaugeMetric]
```

Comment:
I think counter is correct.

Reply:
`CounterMetric` gives per/sec output from `dotnet-counters monitor` so mostly displays zero since blocks are every 12sec; `GaugeMetric` gives totals so far

@Scooletz

Reply:
CounterMetric gives per/sec output from dotnet-counters monitor so mostly displays zero since blocks are every 12sec; GaugeMetric gives totals so far

@Scooletz

e.g. this is output for Gauge


```
[Nethermind.Evm]
    BlockhashOpcode                                                    2,995
    Bn254AddPrecompile                                                 5,930
    Bn254MulPrecompile                                                 6,101
    Bn254PairingPrecompile                                               341
    Calls                                                          2,943,159
    EcRecoverPrecompile                                               57,431
    EvmExceptions                                                          1
    ExtCodeSizeOpcode                                              1,263,899
    ExtCodeSizeOptimizedEQ                                               254
    ExtCodeSizeOptimizedGT                                             2,982
    ExtCodeSizeOptimizedIsZero                                     1,223,772
    ModExpOpcode                                                   1,404,124
    ModExpPrecompile                                                   4,295
    PointEvaluationPrecompile                                              0
    Ripemd160Precompile                                                    0
    SelfDestructs                                                        226
    Sha256Precompile                                                  19,633
    SloadOpcode                                                    9,592,425
    SstoreOpcode                                                   3,705,459
    TloadOpcode                                                            0
    TstoreOpcode                                                           0
```

e.g. this is output for Counter
```
[Nethermind.Evm]
    BlockhashOpcode (Count / 1 sec)                                        0
    Bn254AddPrecompile (Count / 1 sec)                                     0
    Bn254MulPrecompile (Count / 1 sec)                                     0
    Bn254PairingPrecompile (Count / 1 sec)                                 0
    Calls (Count / 1 sec)                                                  0
    EcRecoverPrecompile (Count / 1 sec)                                    0
    EvmExceptions (Count / 1 sec)                                          0
    ExtCodeSizeOpcode (Count / 1 sec)                                      0
    ExtCodeSizeOptimizedEQ (Count / 1 sec)                                 0
    ExtCodeSizeOptimizedGT (Count / 1 sec)                                 0
    ExtCodeSizeOptimizedIsZero (Count / 1 sec)                             0
    ModExpOpcode (Count / 1 sec)                                           0
    ModExpPrecompile (Count / 1 sec)                                       0
    PointEvaluationPrecompile (Count / 1 sec)                              0
    Ripemd160Precompile (Count / 1 sec)                                    0
    SelfDestructs (Count / 1 sec)                                          0
    Sha256Precompile (Count / 1 sec)                                       0
    SloadOpcode (Count / 1 sec)                                            0
    SstoreOpcode (Count / 1 sec)                                           0
    TloadOpcode (Count / 1 sec)                                            0
    TstoreOpcode (Count / 1 sec)                                           0
```

.csv for Gauge (metric)

<img width="505" alt="image" src="https://user-images.githubusercontent.com/1142958/237021462-d6363118-ff36-45a9-84ce-df58215aeebc.png">


.csv for Counter (rate)

<img width="548" alt="image" src="https://user-images.githubusercontent.com/1142958/237021573-6970c265-e526-43c9-9421-22bb20ca0f5c.png">

Reply:
I don't know how dotnet counter works, I assume similarly with prometheus. With prometheus, the `rate` operator result in per-second value, but you can total it up with `increase` operator. Eg: `increase(metric[1m])` total for a minute `increase(metric[$__range])` total for the whole graph. Gauge is for something whose number of observation does not matter as much, only the final value at the end of the time range, and can increase or decrease, eg: Memory usage. Another example, num of operation *per* block. For num of operation (does not matter the range), it should be just counter. The operators does not care if the metric is counter or gauge, it'll just do it, but they are kinda hint to prometheus.